java2typescript-jackson/src/main/java/java2typescript/jackson/module/grammar/MapType.java

```diff
@@ -29,7 +29,11 @@ public MapType() {
 
 	@Override
 	public void write(Writer writer) throws IOException {
-		writer.write("{ [key: ");
+		if (keyType instanceof EnumType) {
+			writer.write("{ [key in ");
```

Comment:
Please add a TODO comment to this line to explain that it should only be used in conjunction with string literal enums.